### ISSUE TITLE
Member invitation flow is confusing

### ISSUE THREAD

**Issue by erikwrede** [graphql-hive/platform#5729]
Consider the following example:
- Send out an invite to someone
- the user clicks the invite but needs to create an account first
- Google SSO is used to create an account
- The user is signed in, but no organizations are visible

You need to open the mail again & click the invite link to accept the invite successfully.

I received the following feedback from multiple colleagues: 
> Erik, I am registered and accepted the invite but can't access the org.

Based on the feedback I received, this flow seems to be counterintuitive. Maybe there is a way to improve it by saving the invitation intent on registration. 

I'm just relaying the feedback; I hope it helps!

**Comment by jdolle** [comment]
This feedback is much appreciated.

Am I correct in understanding that the invite and registration flow isn't the issue but there may be a bug with creating an account or maybe just SSO that causes the invitation intent to be lost?

**Comment by erikwrede** [comment]
@jdolle exactly, it would be great to have the intent saved so it is restored after the sso step

**Comment by jasonkuhrt** [comment]
I have conducted some repros.

### Before All

Each repro begins with:

1. Receive invitation link
   e.g. https://app.graphql-hive.com/join/abc123
3. Not logged in. Redirect to sign in/up
   e.g. https://app.graphql-hive.com/auth/sign-in?redirectToPath=%2Fjoin%2Fabc123

### Flow 1: Sign Up

1. Click “Sign in with Google”, goes to account selection.
2. Select Google account… Sign in with Google account… Redirects to continue Hive sign up, aka. OAuth Authorize Hive.
3. Click “Authorize”. Redirects to Hive. Hive creates its own user resource etc., logs user in, redirects user to:
   https://app.graphql-hive.com/org/new

### Flow 2: Sign In

1. Click “Sign in with Google”, goes to “Choose an account”, choose account, redirect to:
   https://app.graphql-hive.com/org/new


----

- If using sign in with email, then everything works fine. That is to be expected because we do not bounce the user to other domains when using that class of identity provider, thus the existing redirect system (e.g. `?redirectToPath=%2Fjoin%2Fabc123`) works.
- Presumably there are more flow variations (e.g. SSO as reported by @erikwrede, sign in with GitHub, etc.).
- Presumably there is a general solution for all flows that are broken due to being redirected to other platforms, e.g. using cookies (as opposed to, say, using OAuth state variable which does not apply to other classes of identity provider (sso, ...)
- The simplest repro I found so far is Flow 2 above. However, AFAIK, we don't support social sign in in local development. If so, this issue is made more difficult to work on (repro, debug, qa, etc.)
